controller/volume_attachment_controller_test.go

```diff
@@ -1,400 +1,400 @@
+	tc.volAttachment.Spec.AttachmentTickets = map[string]*longhorn.AttachmentTicket{
+		"attachment-01": &longhorn.AttachmentTicket{
+			ID:         "attachment-01",
+			Type:       longhorn.AttacherTypeCSIAttacher,
+			NodeID:     TestNode1,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+	}
+	tc.vol.Status.OwnerID = TestNode1
+	tc.vol.Spec.NodeID = TestNode1
+	tc.vol.Status.CurrentNodeID = TestNode1
+	tc.vol.Status.State = longhorn.VolumeStateAttached
+	tc.copyCurrentToExpect()
+	tc.expectedVolAttachment.Status.AttachmentTicketStatuses = map[string]*longhorn.AttachmentTicketStatus{
+		"attachment-01": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-01",
+			Satisfied: true,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusTrue, "", ""),
+			Generation: 0,
+		},
+	}
+	testCases["test case 4: attach: successfully attached case"] = tc
+	///////////////////////////////////////////////////////////////////
+
+	///////////////////////////////////////////////////////////////////
+	tc = generateVolumeAttachmentTestCaseTemplate(TestVolumeName)
+	tc.volAttachment.Spec.AttachmentTickets = map[string]*longhorn.AttachmentTicket{
+		"attachment-01": &longhorn.AttachmentTicket{
+			ID:     "attachment-01",
+			Type:   longhorn.AttacherTypeCSIAttacher,
+			NodeID: TestNode1,
+			Parameters: map[string]string{
+				"disableFrontend": "true",
+			},
+			Generation: 0,
+		},
+		"attachment-02": &longhorn.AttachmentTicket{
+			ID:         "attachment-02",
+			Type:       longhorn.AttacherTypeSnapshotController,
+			NodeID:     TestNode1,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+	}
+	tc.vol.Status.OwnerID = TestNode1
+	tc.vol.Spec.NodeID = TestNode1
+	tc.vol.Status.CurrentNodeID = TestNode1
+	tc.vol.Status.State = longhorn.VolumeStateAttached
+	tc.copyCurrentToExpect()
+	tc.expectedVolAttachment.Status.AttachmentTicketStatuses = map[string]*longhorn.AttachmentTicketStatus{
+		"attachment-01": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-01",
+			Satisfied: false,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusFalse,
+				longhorn.AttachmentStatusConditionReasonAttachedWithIncompatibleParameters,
+				fmt.Sprintf("volume %v has already attached to node %v with incompatible parameters", tc.vol.Name, tc.vol.Status.CurrentNodeID)),
+			Generation: 0,
+		},
+		"attachment-02": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-02",
+			Satisfied: true,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusTrue, "", ""),
+			Generation: 0,
+		},
+	}
+	testCases["test case 5: attach: fail to attach because the volume is already attached with incompatible parameters"] = tc
```

Comment:
Question: Will the VA controller re-attach the volume for CSIAttacher ticket here?

Reply:
When the volume CR has some changes, VA controller will retry

Reply:
Sorry... Are there any changes in the volume CR...?

Reply:
Sorry for the confusion.

So the CSI ticket cannot be satisfied yet because the volume is already attached with a different parameter. When the volume is detached, the VA will watch that event and retry to fulfill the CSI ticket. Does it answer the question?

Reply:
So in this case the VA controller will do nothing for the CSI Attacher ticket even if it holds a higher priority?

Reply:
Oh. it will. VA controller will set vol.Spec.NodeId to "" to detach the volume

Reply:
Yes, you are right. This seems to be an unexpected behavior when the low priority ticket already attached and the higher priority ticket is created with the same node and different parameter